File: 004.exs
# Juan Vazquez 
# https://github.com/javazquez
# http://javazquez.com
# Elixir Solution to Project Euler problem 4
# http://projecteuler.net/problem=4

defmodule Euler do
  def palindroms(vals) do
    lc x inlist vals,
      y inlist vals, to_string(x*y) === String.reverse(to_string(x*y)) do
      x*y
    end	
  end
end

Euler.palindroms(Enum.to_list(100..999)) |> Enum.max |> IO.puts
#906609

File: 004.clj
;;; Juan Vazquez 
;;; https://github.com/javazquez
;;; http://javazquez.com
;;; Clojure Solution to Project Euler problem 4
;;; http://projecteuler.net/problem=4

(defn isPalindrom?[value](= (str value ) (apply str (reverse  (str value)))))

 (def gen-prod
	(loop[x 999 y 999 acc '()]
	  (if (< x 100)  acc
	    (let [nacc (if (isPalindrom?(* x y)) (cons (* x y) acc) acc)]
              (if (< y 100)
	        (recur (dec x) 999 nacc)
		(recur x (dec y) nacc))))))

;run it
 (reduce max gen-prod)
;user =>906609				
